### PR TITLE
fix: [PIE-8519]: Preventing resetting of storeType while creating Remote pipeline

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
Pipeline storeType should be set as INLINE while creating only one time. Issue was, storeType was getting reset while editing the details in createModal while creating.
Have seen regression for this specific use case in past too as adding the test for the same flow.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/__tests__/PipelineCreate.test.tsx [hunk 1]
```diff
@@ -180,6 +238,38 @@ describe('PipelineCreate test', () => {
...
+        path={remoteTestPath}
+        pathParams={pathParams}
+        queryParams={remoteQueryParams}
+        defaultAppStoreValues={{ isGitSyncEnabled: false, supportingGitSimplification: true }}
+      >
+        <PipelineCreate {...getEditPropsForRemotePipeline()} />
+      </TestWrapper>
+    )
+    await waitFor(() => getByText('continue'))
+    expect(container).toMatchSnapshot()
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/components/PipelineStudio/__tests__/PipelineCreate.test.tsx:253]: can we avoid snapshots and use assertions instead?

Comment (sunnykesh-harness) [reply src/modules/70-pipeline/components/PipelineStudio/__tests__/PipelineCreate.test.tsx:253]: Have already put few assertions as well for this use case. Added snapshots as extra check to verify UI for entire form selected card, connector, repo, branch, filePath etc.

**Comment by abhinavrastogi-harness** [comment]
As discussed, please make a separate PR for replacing snapshots with assertions.

File: src/modules/70-pipeline/components/PipelineStudio/__tests__/PipelineCreate.test.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/70-pipeline/components/PipelineStudio/__tests__/PipelineCreate.test.tsx:253]: Thanks, removing this snapshot in next PR.

**Pull request merged**